File: cases/taint/scala/direct-positive/DirectFlow.scala
package dataflowbench

object DirectFlow {
  def dfb_source(): String = { // DFB-SOURCE: direct-input
    "tainted"
  }

  def dfb_sink(value: String): Unit = {} // DFB-SINK: direct-sink

  def run(): Unit = {
    dfb_sink(dfb_source())
  }
}

File: cases/taint/kotlin/direct-positive/DirectFlow.kt
package dataflowbench

object DirectFlow {
    fun dfb_source(): String { // DFB-SOURCE: direct-input
        return "tainted"
    }

    fun dfb_sink(value: String) {} // DFB-SINK: direct-sink

    fun run() {
        dfb_sink(dfb_source())
    }
}

